File: Codes/009_isPalindrome/9.回文数.js
/*
 * @lc app=leetcode.cn id=9 lang=javascript
 *
 * [9] 回文数
 */

// @lc code=start
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x<0){
        return false
    }

    var str_x=x.toString();
    var i=str_x.length-1;
    var j=0;

    while(j<=i){
        if(str_x[i]!=str_x[j]){
            return false
        }
        j++;
        i--;
    }

    return true;
};
// @lc code=end

File: Codes/009_isPalindrome/9.回文数.go
/*
 * @lc app=leetcode.cn id=9 lang=golang
 *
 * [9] 回文数
 */

// @lc code=start
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	str_x := strconv.Itoa(x)
	i := 0
	j := len(str_x) - 1
	for {
		if str_x[i] != str_x[j] {
			return false
		}
		i++
		j--
		if i > j {
			return true
		}
	}

}

// @lc code=end

